spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -250,6 +247,9 @@ else if (payload instanceof String) {
 	protected void publish(String topic, Object mqttMessage, Message<?> message) {
 		Assert.isInstanceOf(MqttMessage.class, mqttMessage, "The 'mqttMessage' must be an instance of 'MqttMessage'");
 		try {
+			if (!this.mqttClient.isConnected()) {
```

Comment:
Would you mind removing that `if (!this.connectionOptions.isAutomaticReconnect()) {` in the ctor since now we do reconnect ourselves.
Or if it is there, then we rely on the auto-reconnect.
Nothing critical for us from now on.